compatibility/compatibility.cpp

```diff
@@ -0,0 +1,341 @@
+
+                        if (item.size() == 0)
+                            valueIter = value.erase(valueIter);
+                        else
+                            ++valueIter;
+                    }
+
+                    if (value.size() == 0)
+                        jsonIter = json.erase(jsonIter);
+                    else
+                        ++jsonIter;
+                }
+                else if (key == "uri")
+                {
+                    json[key] = std::regex_replace(value.get<std::string>(), std::regex(R"(samples\/[^\/]+)"), KIT_FOLDER);
+                    json[key] = std::regex_replace(value.get<std::string>(), std::regex(R"(\.ts$)"), ".cpp");
+                    ++jsonIter;
+                }
+                else
+                    ++jsonIter;
+            }
+        }
+
+        void SanitizeActualJson(nlohmann::json& json)
+        {
+            for (auto jsonIter = json.begin(); jsonIter != json.end();)
+            {
+                auto& key = jsonIter.key();
+                auto& value = jsonIter.value();
+
+                if (key == "parameterTypeName" && value.get<std::string>().empty())
+                    jsonIter = json.erase(jsonIter);
+                else if (key == "exception")
+                    jsonIter = json.erase(jsonIter);
+                else if (key == "message")
+                    jsonIter = json.erase(jsonIter);
+                else if (key == "line")
+                    jsonIter = json.erase(jsonIter);
+                else if (key == "snippets")
+                    jsonIter = json.erase(jsonIter);
+                else if (value.is_object())
+                {
+                    SanitizeActualJson(value);
+                    if (value.size() == 0)
+                        jsonIter = json.erase(jsonIter);
+                    else
+                        ++jsonIter;
+                }
+                else if (value.is_array())
+                {
+                    auto idx = 0;
+                    for (auto valueIter = value.begin(); valueIter != value.end();)
+                    {
+                        auto& item = *valueIter;
+
+                        if (item.is_object())
+                            SanitizeActualJson(item);
+
+                        if (item.size() == 0)
+                            valueIter = value.erase(valueIter);
+                        else
+                            ++valueIter;
+                    }
+
+                    if (value.size() == 0)
+                        jsonIter = json.erase(jsonIter);
+                    else
+                        ++jsonIter;
+                }
+                else
+                    ++jsonIter;
+            }
+        }
+
+        struct BroadcastListener
+        {
+            BroadcastListener(std::filesystem::path ndjsonin, std::filesystem::path expectedndjson, std::filesystem::path ndout, cucumber_cpp::library::util::Broadcaster& broadcaster)
```

Comment:
What is the design decision of having expectedndjson separate from ndjsonin from compatibility?